### DISCUSSION TITLE
请教yt-dlp使用问题

### DISCUSSION THREAD

**Discussion by gzgzcyq** [1715173329/UnblockNeteaseMusic#459] [Q&A] [answered]
想麻烦请教一下，yt-dlp相关问题
### 1.自己的使用环境
node 16.1.0
启动命令 node app.js
![image](https://user-images.githubusercontent.com/38951166/149476531-1aff09ca-a85c-418d-9f69-fa662609a7ee.png)
直接将yt-dlp.exe和app.js放置在了同一层目录下
![image](https://user-images.githubusercontent.com/38951166/149477558-c989ea0f-0499-4143-8502-5a176fc84434.png)

### 2.遇到的问题
如果音源需要使用yt-dlp就会报错，如图
![image](https://user-images.githubusercontent.com/38951166/149478663-d5947db5-ac34-4145-80f5-f40c70c3ddba.png)

### 3.想要请教的问题
1.yt-dlp这样使用是否正确？

### 补充：自己已经做过的尝试
1.复制音源url出来直接调用yt-dlp下载，可以成功下载；
![image](https://user-images.githubusercontent.com/38951166/149480258-001c9180-58a8-47b1-9cc4-94b7e8bb1e14.png)
2.看了issue自己之前的node是14，切换到16还是不行；

首先要感谢各位大佬的维护，如能帮忙解答，万分感谢

**Comment by pan93412** [comment, marked as answer]
應該放在你 PATH 環境變數指定的位置。

**Reply by gzgzcyq** [reply]
刚刚忘了写，尝试过配置Path环境变量，还是会报错
![image](https://user-images.githubusercontent.com/38951166/149486175-d2a8e53b-fa1d-4852-9af7-4ec4e4ada2fe.png)

这个是在其他位置调用yt-dlp
![image](https://user-images.githubusercontent.com/38951166/149486368-878ae3d6-529f-4c5f-99fa-fb44fe300582.png)

**Reply by pan93412** [reply]
我看錯了，看起來好像是 `yt-dlp` 的輸出不尋常。你能手動呼叫 `(spawn)` 那一行的命令嗎？

**Reply by gzgzcyq** [reply]
实在抱歉，不是很明白您的意思。
‘呼叫（spawn）’不是很明白这句什么意思

**Reply by pan93412** [reply]
> 实在抱歉，不是很明白您的意思。
> ‘呼叫（spawn）’不是很明白这句什么意思

在 cmd 運行 `yt-dlp -f 140 --dump-json ytsearch1:花海 - 周杰伦`

**Reply by gzgzcyq** [reply]
很长一段，我截图了开头和结尾
![image](https://user-images.githubusercontent.com/38951166/149488105-e3a17c95-b919-4c55-ab7f-beb6ff55966f.png)


![image](https://user-images.githubusercontent.com/38951166/149488048-6bc70750-c682-4882-909e-840c8200333c.png)

**Reply by pan93412** [reply]
應該是 UNM 的 bug。我現在修。

**Reply by gzgzcyq** [reply]
刚刚发现，可能是因为自己开了代理的关系，关闭了以后没有报错了。
![image](https://user-images.githubusercontent.com/38951166/149488841-391bf3be-61c0-4626-a251-0a1c1ce9ff89.png)

![image](https://user-images.githubusercontent.com/38951166/149488919-cc06a678-b4bd-4bcd-8b9a-54ad2fae46fa.png)

**Reply by pan93412** [reply]
喔，那就好 👍

**Reply by gzgzcyq** [reply]
感谢解答 谢谢
